### PR TITLE
Feat/home screen presentation

### PR DESCRIPTION
**Pull request by ParnaChatterjee-Grid**

### CODE REVIEW

File: data/src/main/java/com/example/data/di/ApolloClient.kt [hunk 1]
```diff
@@ -2,26 +2,22 @@ package com.example.data.di
...
-        return ApolloClient.Builder().serverUrl(RickMortConstants.RICK_MORTY_URL)
+        return ApolloClient.Builder().serverUrl(com.example.data.RickMortConstants.RICK_MORTY_URL)
             .okHttpClient(httpClient)
             .build()
     }
 
-    @Provides
-    @Singleton
+    @dagger.Provides
+    @javax.inject.Singleton
```
Comment (shreyansh-mm) [inline data/src/main/java/com/example/data/di/ApolloClient.kt]: Why are we removing imports?

File: data/src/main/java/com/example/data/repository/CharacterRepositoryImpl.kt [hunk 2]
```diff
@@ -40,8 +40,12 @@ class CharacterRepositoryImpl @Inject constructor(private val apolloClient: Apol
...
-
         }
     }
 }
 
+
+
+
+
+
```
Comment (shreyansh-mm) [inline data/src/main/java/com/example/data/repository/CharacterRepositoryImpl.kt]: Let's remove extra lines if we don't need them

File: presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+            else -> {
+                Text(
+                    text = " No data found ",
+                    color = MaterialTheme.colorScheme.error,
+                    modifier = Modifier
+                )
+            }
+        }
+
```
Comment (shreyansh-mm) [inline presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt]: Let's discuss this over the call

File: presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+
+
+    }
+}
+
+@Composable
+private fun SetCharactersList(characterList: kotlin.collections.List<com.example.domain.models.Characters>) {
```
Comment (shreyansh-mm) [inline presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt]: Add imports?

File: presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    LazyVerticalGrid(
+        columns = androidx.compose.foundation.lazy.grid.GridCells.Fixed(2),
```
Comment (shreyansh-mm) [inline presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt]: Same

File: presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        modifier = Modifier.fillMaxWidth(),
+        contentPadding = androidx.compose.foundation.layout.PaddingValues(all = 16.dp),
```
Comment (shreyansh-mm) [inline presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt]: Same

File: presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+                modifier = Modifier
+                    .fillMaxSize()
+                    .aspectRatio(1f)
+            )
+            Text(
+                text = character.name,
+                modifier = Modifier
+                    .padding(8.dp)
+                    .align(androidx.compose.ui.Alignment.BottomCenter),
+                color = androidx.compose.ui.graphics.Color.White,
```
Comment (shreyansh-mm) [inline presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt]: Same

File: presentation/src/main/java/com/example/presentation/viewmodels/CharacterViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    val charactersState: StateFlow<ResultState<List<Characters?>>> = _charactersState
+
+    init{
+        getAllCharacters()
+    }
+
+
+    private fun getAllCharacters() {
+        _charactersState.value = ResultState.Loading
+        viewModelScope.launch(IO) {
```
Comment (shreyansh-mm) [inline presentation/src/main/java/com/example/presentation/viewmodels/CharacterViewModel.kt]: Dispatcher should not be used like this

File: presentation/src/main/java/com/example/presentation/viewmodels/ViewModelModule.kt [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+import javax.inject.Singleton
+
+@Module
+class ViewModelModule{
+    @Provides
+    @Singleton
+
+    fun provideViewModelFactory(getCharactersUsecase : GetCharactersUsecase): ViewModelfactory {
+        return ViewModelfactory(getCharactersUsecase)
+    }
```
Comment (shreyansh-mm) [inline presentation/src/main/java/com/example/presentation/viewmodels/ViewModelModule.kt]: Fix spacing

**Review by shreyansh-mm — CHANGES_REQUESTED** [review]

File: presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt (hunk 5 above)
Comment (ParnaChatterjee-Grid) [reply presentation/src/main/java/com/example/presentation/screens/HomeScreen.kt]: Android Studio taking automatic imports like this for me.

**Review by shreyansh-mm — APPROVED** [review]

**Pull request merged**